### PR TITLE
Overhaul the way we vectorize records and find duplicates 

### PR DESCRIPTION
**Pull request by JS-BC**
- `GetRecordDuplicates` now accepts multiple recordIDs instead of one
- The returned object now returns the PrimaryKeyValues for each duplicate record
- Created a `VectorBase` class that holds many of the functions shared between the vector dupe and vector sync packages
- Vector sync package can now be configured to vectorize many records at once in batches
- Associated meta data for each vector is more sensible
- A vector's metadata now also stores its Primary Key Values
- When searching for record dupes, if an entity document is not found, we create one
 - The template generated just uses all fields of the given entity 
 - The AI model and vector DB used is the first one returned from their respective lists
- When vectorizing records, if a vector index doesnt exist we create one
  - The AI model and vector DB used is the first one returned from their respective lists
 - Added Vector-Pinecone and Ai-Mistral package dependencies to MJServer as those are our default embedding model and vector DB
 - Added a new `EmbedTexts` function to the Embedding class
 - Few more changes but those are the major ones 

Once the frontend receives the list of duplicates, It will then need to use the key value pairs to fetch and load the underlying entity record and display it

### CODE REVIEW

**Comment by AN-BC** [comment]
@JS-BC - reviewed the code. No real comments, I think we should merge this and then add the code for the spMatch call separate so we can easily review that separately after @cadam11  has a chance to look this over.

Also, for ALL places in the UI where we were previously doing manual key concatenation and parsing like in the UI where we did that with the Key|Value||Key2|Value2 and so on, have you fixed that to always use functions in the CompositeKey class. I think maybe that is still pending??

**Comment by JS-BC** [comment]
> @JS-BC - reviewed the code. No real comments, I think we should merge this and then add the code for the spMatch call separate so we can easily review that separately after @cadam11 has a chance to look this over.
> 
> Also, for ALL places in the UI where we were previously doing manual key concatenation and parsing like in the UI where we did that with the Key|Value||Key2|Value2 and so on, have you fixed that to always use functions in the CompositeKey class. I think maybe that is still pending??

Im changing all of those instances and yes that work is still pending

File: packages/AI/Mistral/src/models/mistralEmbedding.ts [hunk 1]
```diff
@@ -15,13 +15,43 @@ export class MistralEmbedding extends Embeddings {
...
+     * Mistral AI embedding endpoint outputs vectors in 1024 dimensions
+     */
+    public async EmbedText(params: EmbedTextParams): Promise<EmbedTextResult> {
+        params.model = params.model || "mistral-embed";
+        const response: EmbeddingResponse = await MistralEmbedding._client.embeddings(params.model, [params.text]);
+        return {
+            object: response.object,
+            model: response.model,
+            ModelUsage: new ModelUsage(response.usage.prompt_tokens, response.usage.completion_tokens),
+            vector: response.data[0].embedding
```
Comment (cadam11) [inline packages/AI/Mistral/src/models/mistralEmbedding.ts:30]: Heads up that `response.data` could be an empty array, which would throw an error when trying to access `.embedding` on undefined. The Typescript compiler can't know this ahead of time, unfortunately

File: packages/AI/Vectors/src/models/VectorBase.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        const EDTemplate: string = entityFields.map(entityField => {
+            return `${entityField.Name}: \$\{${entityField.Name}\}`;
+        }).join(" ");
+
+        const md: Metadata = new Metadata();
+        const entityDocument: EntityDocumentEntity = await md.GetEntityObject<EntityDocumentEntity>("Entity Documents");
+        entityDocument.NewRecord();
+        entityDocument.Set("Name", `Default duplicate record Entity Document for the ${entity.Name} entity using vector database ${vectorDatabase.Name} and AI Model ${AIModel.Name}`);
+        entityDocument.Set("EntityID", entityID);
+        entityDocument.Set("TypeID", 9); //hardcoded, but we know that 9 is the ID for Record Duplicates
```
Comment (cadam11) [inline packages/AI/Vectors/src/models/VectorBase.ts:161]: Might want to consider adding a `constants.ts` with `export RECORD_DUPLICATES_TYPE_ID=9;` just for easier maintainability down the road

**Review by cadam11 — APPROVED** [review]
LGTM!

**Pull request merged**